godef not working with type alias

When I use type alias with golang 1.9, I got error:

```
godef -t -i -f source.go -o 121
godef: no identifier found
```
source file:
```go
type RedisOption = redis.Options
```

Or error:

```
godef -t -i -f source.go -o 594
godef: no declaration found for pkg.RedisOption
```

Hey, I always meet this problem, and ijc's pr can't effect in my scene.

Problem source: https://github.com/libp2p/go-libp2p-examples
Use command: godef -debug -f libp2p-host/host.go libp2p.New

error:
src/github.com/libp2p/go-libp2p/libp2p.go:12:13: expected type, found '='

--- a/go/ast/ast.go
+++ b/go/ast/ast.go
@@ -772,6 +772,7 @@ type (
 	TypeSpec struct {
 		Doc     *CommentGroup // associated documentation; or nil
 		Name    *Ident        // type name
+		Assign  token.Pos     // position of '=', if any
 		Type    Expr          // *Ident, *ParenExpr, *SelectorExpr, *StarExpr, or any of the *XxxTypes
 		Comment *CommentGroup // line comments; or nil
 	}

--- a/go/parser/parser.go
+++ b/go/parser/parser.go
@@ -2027,8 +2027,12 @@ func parseTypeSpec(p *parser, doc *ast.CommentGroup, decl *ast.GenDecl, _ int) a
 	// at the identifier in the TypeSpec and ends at the end of the innermost
 	// containing block.
 	// (Global identifiers are resolved in a separate phase after parsing.)
-	spec := &ast.TypeSpec{doc, ident, nil, p.lineComment}
+	spec := &ast.TypeSpec{doc, ident, token.NoPos, nil, p.lineComment}
 	p.declare(spec, p.topScope, ast.Typ, ident)
+	if p.tok == token.ASSIGN {
+		spec.Assign = p.pos
+		p.next()
+	}
 	typ := p.parseType()
 	p.expectSemi() // call before accessing p.linecomment
 	spec.Type = typ

--- a/go/parser/parser_test.go
+++ b/go/parser/parser_test.go
@@ -46,6 +46,7 @@ var validPrograms = []interface{}{
 	`package p; type T []int; var a []bool; func f() { if a[T{42}[0]] {} };`,
 	`package p; type T []int; func g(int) bool { return true }; func f() { if g(T{42}[0]) {} };`,
 	`package p; type T []int; func f() { for _ = range []int{T{42}[0]} {} };`,
+	`package p; type T []int; type U = T; func f() int { return U{42}[0] };`,
 	`package p; var a = T{{1, 2}, {3, 4}}`,
 	`package p; func f() { select { case <- c: case c <- d: case c <- <- d: case <-c <- d: } };`,
 	`package p; func f() { if ; true {} };`,

--- a/go/types/types.go
+++ b/go/types/types.go
@@ -230,8 +230,8 @@ func (ctxt *exprTypeContext) exprType(n ast.Node, expectTuple bool, pkg string) 
 		if obj == nil || obj.Kind == ast.Bad {
 			break
 		}
-		// A type object represents itself.
-		if obj.Kind == ast.Typ {
+		// A non-aliased type object represents itself.
+		if obj.Kind == ast.Typ && !isTypeAlias(obj) {
 			// Objects in the universal scope don't live
 			// in any package.
 			if parser.Universe.Lookup(obj.Name) == obj {
@@ -869,6 +869,14 @@ func containsNode(node, x ast.Node) (found bool) {
 
 func isNamedType(typ Type) bool {
 	return typ.Underlying(false).Node != typ.Node
+}
+
+func isTypeAlias(obj *ast.Object) bool {
+	if obj.Kind != ast.Typ {
+		return false
+	}
+	ts, ok := obj.Decl.(*ast.TypeSpec)
+	return ok && ts.Assign.IsValid()
 }
 
 func fields2type(fields *ast.FieldList) ast.Node {

